File: Siyeon/[02_17] 그리디/체육복.cpp
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        answer = n-lost.length;
        
        for(int i=0; i<lost.length; i++) {
            for(int j=0; j<reserve.length; j++) {
                if(lost[i]==reserve[j]) {
                    reserve[j] = -2;
                    lost[i] = -4;
                    answer++;
                }
            }
        }
        
        for(int i=0; i<lost.length; i++) {
            if(lost[i]==-4) continue;
            
            for(int j=0; j<reserve.length; j++) {
                if(lost[i]==reserve[j]+1) {
                    reserve[j] = -2;
                    if(answer<n) answer++;
                }
                else if(lost[i]==reserve[j]-1) {
                    reserve[j] = -2;
                    if(answer<n) answer++;
                }
            }
        }
        
        return answer;
    }
}

File: Ayeon/[2월 17일] 그리디 알고리즘/체육복.py
def solution(n, lost, reserve):
    same=[]
    for r in reserve:
        if r in lost:
            same.append(r)
    for s in same:
        reserve.remove(s)
        lost.remove(s)
    answer = n - len(lost)  
    for r in reserve:
        if (r-1 in lost) | (r+1 in lost):
            answer+=1       
    if answer > n:
        answer = n
    return answer